### PR TITLE
fountain: check for empty parts in validation

### PR DESCRIPTION
**Pull request by jeandudey**

### CODE REVIEW

File: src/fountain.rs [hunk 1]
```diff
@@ -861,7 +861,7 @@ mod tests {
     }
 
     #[test]
-    fn test_empty_decoder() {
+    fn test_empty_encoder() {
```
Comment (dspicher) [inline src/fountain.rs:864]: Good catch!

File: src/fountain.rs [hunk 2]
```diff
@@ -423,18 +432,18 @@ impl Decoder {
...
-        if part.message_length != self.message_length {
-            return false;
-        }
-        if part.checksum != self.checksum {
+        if self.received.is_empty() {
             return false;
         }
-        if part.data.len() != self.fragment_length {
+
+        if part.sequence_count != self.sequence_count
```
Comment (dspicher) [inline src/fountain.rs:439]: Can you explain the intention with this diff? This is only reshuffling stuff, right?

If you want to keep it, can you please put it into a separate commit, because it seems unrelated to the commit message (unless I didn't get something).

File: src/fountain.rs [hunk 3]
```diff
@@ -925,6 +934,21 @@ mod tests {
         assert!(!decoder.validate(&part));
     }
 
+    #[test]
+    fn test_empty_decoder_empty_part() {
+        let mut decoder = Decoder::default();
+        let part = Part {
+            sequence: 0,
```
Comment (dspicher) [inline src/fountain.rs:941]: This is a tiny bit misleading, as only one of the empty fields here will be tested.

But we can leave this for a future cleanup.

**Review by dspicher — CHANGES_REQUESTED** [review]

File: src/fountain.rs (hunk 2 above)
Comment (jeandudey) [reply src/fountain.rs:439]: Just to keep the same formatting between the if blocks as the if/returns block would have been much bigger when adding the other equalities, so that's the reason for the same commit.

Comment (jeandudey) [reply src/fountain.rs:439]: Added an extra commit for this one and for the check

File: src/fountain.rs (hunk 3 above)
Comment (jeandudey) [reply src/fountain.rs]: Added a couple more checks

**Comment by jeandudey** [comment]
@dspicher updated the PR.

File: src/fountain.rs [hunk 4]
```diff
@@ -930,6 +939,45 @@ mod tests {
         assert!(!decoder.validate(&part));
     }
 
+    #[test]
+    fn test_empty_decoder_empty_part() {
```
Comment (dspicher) [inline src/fountain.rs:938]: Perfect, thank you!

**Review by dspicher — APPROVED** [review]
